orcestra/meteor.py

```diff
@@ -0,0 +1,83 @@
+import getpass
+import json
+import pathlib
+from datetime import datetime
+from io import BytesIO
+
+import numpy as np
+import pandas as pd
+import requests
+
+
+def get_planet_credentials():
+    """Get credentials for PLANET authentication.
+
+    User name and password can be stored in a file ~/config/orcestra/planet.json as:
+
+    {
+        "user": <username>,
+        "password": <password>
+    }
+
+    Otherwise, the user is prompted for input.
+    """
+    passwd_file = pathlib.Path("~/.config/orcestra/planet.json").expanduser().resolve()
+
+    if passwd_file.is_file():
+        with passwd_file.open("r") as fp:
+            j = json.loads(fp.read())
+
+            return j["user"], j["password"]
+
+    return input("Username: "), getpass.getpass("Password: ")
+
+
+def request_telemetry(user, password, mission_id="PERCUSION", vehicle_id="METEOR"):
+    """Retrieve telemetry information from the PLANET system."""
+
+    # Retrieve METEOR position via PLANET API
+    url = f"https://dlr.atmosphere.aero/api/v1/telemetry/{mission_id}/"
+    query_parameters = {
+        "creator": vehicle_id,
+        "fmt": "csv",
+    }
+
+    r = requests.get(url, params=query_parameters, auth=(user, password))
+    r.raise_for_status()
+
+    return r
+
+
+def get_meteor_track():
+    """Request the METEOR track from PLANET and return as xr.Dataset."""
+    # Retrieve telemetry data for METEOR
+    user, password = get_planet_credentials()
+    r = request_telemetry(user, password)
+
+    # Read CSV response as pandas dataframe
+    ds = pd.read_csv(BytesIO(r.text.encode())).to_xarray()
+
+    # Sanitise dataset
+    ds = (
+        ds[["issued", "position.0", "position.1"]]
+        .drop_vars(
+            "index",
+        )
+        .rename(
+            {
+                "index": "time",
+                "issued": "time",
+                "position.0": "lon",
+                "position.1": "lat",
+            }
+        )
+    )
+
+    # Properly parse datetimes
+    ds = ds.assign_coords(time=[datetime.fromisoformat(str(t)) for t in ds.time.values])
+
+    # Remove duplicate lat/lon information
+    pos = [(lon, lat) for lon, lat in zip(ds.lon.values, ds.lat.values)]
+    _, idx = np.unique(pos, axis=0, return_index=True)
```

Comment:
This could possibly delete revisited positions, i. e. Meteor leaves a station and returns back to the same station at a later time. Due to GPS based navigation, this might actually not be a too unlikely case.
I'd suggest removing identical locations only if they are consecutive (and probably also only if their timestamps match).

Reply:
Oh yeah, I hadn't thought of that! The new implementation is based on the duplicates in the **difference** of lat and lon. And apparently it keeps a fair amount of additional points in the dataset... 😬

Reply:
I am not completely sure what the timestep in the METEOR telemetry means. It seems like every single lat/lon pair is repeated several times, which looks like the PLANET database is updated more often than the actual ship location. This is what I tried to sanitise here... but we could also keep also time steps (about 45,000 instead of 2,500)